### PR TITLE
Add correct transitions to menu items

### PR DESCRIPTION
**Pull request by ddouglasz**
#### Summary

Add correct transitions to menu items

https://github.com/commercetools/merchant-center-application-kit/assets/19975842/0740cad9-c5b1-4f4f-a8a1-5a7433fa8bdf

### CODE REVIEW

**Comment by chloe0592** [comment]
@ddouglasz thank you for working on it! 🙌 
I think the `menu-item` height changes for 4px when hovered but I'm not sure if that should be the case.
But maybe @FilPob can let us know?

Screenshot 2023-09-19 at 09 04 09

**Comment by FilPob** [comment]
yeah there is this small unwanted jump. The menu item's height should not be affected. Only the size of the icon

**Comment by chloe0592** [comment]
One more question @FilPob, is there some transition for background color on hover? It seems to me that's the case from the prototype but I cannot find details in design for it.

https://github.com/commercetools/merchant-center-application-kit/assets/52276952/607f2c31-3f4e-4b0d-8225-33d0c71fb39f

**Comment by ddouglasz** [comment]
> One more question @FilPob, is there some transition for background color on hover? It seems to me that's the case from the prototype but I cannot find details in design for it.
> 
>  Screen.Recording.2023-09-19.at.09.02.23.mov

Yes, true. I think there is a transition for the background too. Thanks for catching that 🙏🏾

**Comment by ddouglasz** [comment]
Jumpie displacement on hover handled here: https://github.com/commercetools/merchant-center-application-kit/pull/3218/commits/fc3bc5b7226c38ce60f7dbbc015f3329413daeea

All other feedback on color transition  and transition on sublist can also be seen on the preview:


https://github.com/commercetools/merchant-center-application-kit/assets/19975842/9635f49c-f2d2-4df5-b769-2d43816f7588

**Comment by emmenko** [comment]
The "zooming in / out" behavior of the main items looks somehow weird to me. @FilPob can we use another effect? For example in the submenu it moves left <--> right 🤔

**Comment by FilPob** [comment]
@ddouglasz  @emmenko  thanks it's looking great so far! I think this "zoom" animation has potential to look really nice, but it's not 100% correct yet, thats why it looks a bit off. The issue is that currently the menu item label also moves a bit down on hover. Instead, it should only move to the right. What should happen on hover is:

- the icons scales up
- the label is pushed to the right as a consequence of the up-scalled icon. 

It is difficult to explain :D but in this [Figma prototype](https://www.figma.com/proto/gC6Yz03ThScI6Mk23cjKDt/Main-Navi2023?page-id=0%3A1&node-id=402-53029&scaling=min-zoom&starting-point-node-id=402%3A53029&show-proto-sidebar=1&mode=design&t=2KW24S0LSbHHYmgS-1) it works as expected and IMO looks quite smooth. 

Let me know if you want to have a short sync :)

**Comment by emmenko** [comment]
Got it, thanks 👌

**Comment by ddouglasz** [comment]
> @ddouglasz @emmenko thanks it's looking great so far! I think this "zoom" animation has potential to look really nice, but it's not 100% correct yet, thats why it looks a bit off. The issue is that currently the menu item label also moves a bit down on hover. Instead, it should only move to the right. What should happen on hover is:
> 
> * the icons scales up
> * the label is pushed to the right as a consequence of the up-scalled icon.
> 
> It is difficult to explain :D but in this [Figma prototype](https://www.figma.com/proto/gC6Yz03ThScI6Mk23cjKDt/Main-Navi2023?page-id=0%3A1&node-id=402-53029&scaling=min-zoom&starting-point-node-id=402%3A53029&show-proto-sidebar=1&mode=design&t=2KW24S0LSbHHYmgS-1) it works as expected and IMO looks quite smooth.
> 
> Let me know if you want to have a short sync :)

I understand you.  I am going over the approach again.
Thank you.

**Comment by ddouglasz** [comment]
@FilPob, updated here: https://github.com/commercetools/merchant-center-application-kit/pull/3218/commits/fdd141460d0f98f52fea016500b44cadff8a669e

Thank you. 


https://github.com/commercetools/merchant-center-application-kit/assets/19975842/078c41d2-aca7-43e5-aa3a-4979ba7b6e1b

**Review by kark — COMMENTED** [review]
Thank you Douglas 🙌
I've noticed 2 minor issues.

File: packages/application-shell/src/components/navbar/navbar-new/navbar.mod.css [hunk 1]
```diff
@@ -87,15 +91,25 @@
 
 .list-item .item-icon-text {
   display: flex;
+  position: absolute;
   flex-direction: row;
   justify-content: center;
   align-items: center;
+  top: var(--spacing-25);
+  left: var(--spacing-25);
 }
```
[the remark this answers is not in the document]
Comment (ddouglasz) [reply packages/application-shell/src/components/navbar/navbar-new/navbar.mod.css:100]: Thank you, Kacper 🙏🏾. I am looking into the issues now.

File: packages/application-shell/src/components/navbar/navbar-new/navbar.mod.css [hunk 2]
```diff
@@ -67,10 +67,14 @@
   cursor: pointer;
 }
 
+.list:not(.item_menu__active) .list-item,
+.list:not(.item__active) .list-item {
+  transition: all 150ms ease-out;
+}
```
[the remark this answers is not in the document]
Comment (kark) [reply packages/application-shell/src/components/navbar/navbar-new/navbar.mod.css:73]: Perhaps this:
```suggestion
.list:not(.item_menu__active) .list-item:not(:hover),
.list:not(.item__active) .list-item:not(:hover) {
  transition: all 150ms ease-out;
}
```
would help here 🤔

**Comment by kark** [comment]
I've noticed one more regression - it appears that not the entire submenu item is clickable now:

Just in case, I'm here to lend a hand if you need

**Comment by ddouglasz** [comment]
> I've noticed one more regression - it appears that not the entire submenu item is clickable now:  [  ](https://user-images.githubusercontent.com/49066275/270364167-6d47e4e9-7818-4101-9722-c43873b9706d.gif) [ ](https://user-images.githubusercontent.com/49066275/270364167-6d47e4e9-7818-4101-9722-c43873b9706d.gif) Just in case, I'm here to lend a hand if you need

Thank you for all the feedback and suggestions. 
So I think this has been the behaviour from main too, I can try to make it fully clickable here too.

**Comment by kark** [comment]
> Thank you for all the feedback and suggestions. So I think this has been the behaviour from main too, I can try to make it fully clickable here too.

Oh, sorry, my bad. Please disregard this then, it has nothing to do with this PR.

File: packages/application-shell/src/components/navbar/navbar-new/navbar.mod.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (ddouglasz) [reply packages/application-shell/src/components/navbar/navbar-new/navbar.mod.css:101]: All feedback updated here: https://github.com/commercetools/merchant-center-application-kit/pull/3218/commits/b7e6ecb49f4b379dd6213ba3055131b2965849b2

Thank you.

File: packages/application-shell/src/components/navbar/navbar-new/navbar.mod.css (hunk 2 above)
[the remark this answers is not in the document]
Comment (ddouglasz) [reply packages/application-shell/src/components/navbar/navbar-new/navbar.mod.css]: updated here: https://github.com/commercetools/merchant-center-application-kit/pull/3218/commits/b7e6ecb49f4b379dd6213ba3055131b2965849b2

Thank you 🙏🏾

**Review by kark — COMMENTED** [review]
Thanks once again for all the improvements! 🙏
Apologies for accidentally closing this PR 😅
I've come across one more issue, although I'm not entirely sure if it's a regression introduced by this PR.
 


@FilPob changes from this PR are deployed [here](https://mc-15303.mc-preview.europe-west1.gcp.escemo.com/main-navigation-test/welcome)

File: packages/application-shell/src/components/navbar/navbar-new/navbar.tsx [hunk 3]
```diff
@@ -112,7 +112,12 @@ export const ApplicationMenu = (props: ApplicationMenuProps) => {
   const menuItemBoundingClientRect =
     menuItemRef.current?.getBoundingClientRect();
   const menuItemTop = menuItemBoundingClientRect?.top || 0;
-  const menuItemBottom = menuItemBoundingClientRect?.bottom || 0;
+  const menuItemBottom = menuItemBoundingClientRect
+    ? menuItemBoundingClientRect.bottom +
+      (props.isMenuOpen
+        ? DIMENSIONS.subMenuItemVerticalOffset / 2
+        : DIMENSIONS.subMenuItemVerticalOffset)
+    : 0;
```
[the remark this answers is not in the document]
Comment (ddouglasz) [reply packages/application-shell/src/components/navbar/navbar-new/navbar.tsx:120]: It's mainly to just compensate for the difference especially checking for when tooltip is present and sublist is above of below. 
The offfset is menuItemHeight for when its beow the tooltip, but when it's above its half the hight. And this offset mostly occurs when the tooltip exists so we are checking for that too.

**Review by kark — APPROVED** [review]
Thank you for your efforts! The changes looks great now!

**Review by chloe0592 — APPROVED** [review]
@ddouglasz beautifully done! ✨ Thank you for all the hard work 🙇‍♀️

File: packages/application-shell/src/components/navbar/navbar-new/navbar.mod.css [hunk 4]
```diff
@@ -117,6 +125,11 @@
   z-index: 1;
 }
 
+.list-item:hover .icon {
+  /* 1.16 is roughly the ratio of icon-size-hover to --icon-size */
+  transform: scale(1.2);
+}
+
```
Comment (chloe0592) [inline packages/application-shell/src/components/navbar/navbar-new/navbar.mod.css:132]: @ddouglasz sorry just noticed one more tiny thing, I think this block is now duplicated to the one in line 102?

**Review by emmenko — APPROVED** [review]
Wonderful! 🥳

**Pull request merged**